processing/src/main/java/org/apache/druid/java/util/metrics/SysMonitorOshi.java

```diff
@@ -0,0 +1,407 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.metrics;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.service.ServiceEmitter;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.CentralProcessor.TickType;
+import oshi.hardware.GlobalMemory;
+import oshi.hardware.HWDiskStore;
+import oshi.hardware.HardwareAbstractionLayer;
+import oshi.hardware.NetworkIF;
+import oshi.hardware.VirtualMemory;
+import oshi.software.os.FileSystem;
+import oshi.software.os.InternetProtocolStats;
+import oshi.software.os.OSFileStore;
+import oshi.software.os.OperatingSystem;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+
+public class SysMonitorOshi extends FeedDefiningMonitor
+{
+  private static final Logger log = new Logger(SysMonitorOshi.class);
+  private final SystemInfo si = new SystemInfo();
+  private final HardwareAbstractionLayer hal = si.getHardware();
+  private final OperatingSystem os = si.getOperatingSystem();
+  private final List<String> fsTypeWhitelist = ImmutableList.of("local");
+  private final List<String> netAddressBlacklist = ImmutableList.of("0.0.0.0", "127.0.0.1");
+  private final List<Stats> statsList;
+
+  private final Map<String, String[]> dimensions;
+
+  public SysMonitorOshi()
+  {
+    this(ImmutableMap.of());
+  }
+
+  public SysMonitorOshi(Map<String, String[]> dimensions)
+  {
+    this(dimensions, DEFAULT_METRICS_FEED);
+  }
+
+  public SysMonitorOshi(Map<String, String[]> dimensions, String feed)
+  {
+    super(feed);
+    Preconditions.checkNotNull(dimensions);
+    this.dimensions = ImmutableMap.copyOf(dimensions);
+
+    this.statsList = new ArrayList<Stats>();
+    this.statsList.addAll(
+        Arrays.asList(
+            new MemStats(),
+            new SwapStats(),
+            new FsStats(),
+            new DiskStats(),
+            new NetStats(),
+            new CpuStats(),
+            new SysStats(),
+            new TcpStats()
+        )
+    );
+  }
+
+  @Override
+  public boolean doMonitor(ServiceEmitter emitter)
+  {
+    for (Stats stats : statsList) {
+      stats.emit(emitter);
+    }
+    return true;
+  }
+
+  private interface Stats
+  {
+    void emit(ServiceEmitter emitter);
+  }
+
+  private class MemStats implements Stats
+  {
+    @Override
+    public void emit(ServiceEmitter emitter)
+    {
+      GlobalMemory mem = hal.getMemory();
+      if (mem != null) {
+        final Map<String, Long> stats = ImmutableMap.of(
+            "sysO/mem/max", mem.getTotal(),
```

Comment:
Yeah, that was my concern. If someone uses SysMonitor till Oshi monitor is fully documented and tested. 
I thought we'll change back to Sys once SysMonitor is deprecated.

Reply:
this will double the volume of metrics and there are tools out there already tracking these metrics. If we change the name, then those tools need to adjust. That will be painful. so let's retain the same names.

Reply:
Sounds correct. Retained